pkg/operator/startupmonitorreadiness/readiness_checks.go

```diff
@@ -128,11 +142,15 @@ func newPodRunning(podClient corev1client.PodInterface, monitorRevision int) fun
 		if len(apiServerPods.Items) == 0 {
 			return false, "PodNotRunning", "unable to check the pod's status, waiting for Kube API server pod to show up"
 		}
-		if len(apiServerPods.Items) != 1 {
-			return false, "PodListError", fmt.Sprintf("unable to check the pod's status, unexpected number of Kube API server pods %d, expected only one pod", len(apiServerPods.Items))
+
+		kasPodFound, kasPod, kasPodErr := findSingleMatchingPod(apiServerPods.Items, currentNodeName)
+		if kasPodErr != nil {
+			return false, "PodListError", fmt.Sprintf("unable to check the pod's status: %v", kasPodErr)
+		}
+		if !kasPodFound {
+			return false, "PodNotFound", fmt.Sprintf("unable to check the pod's status, haven't found a pod that would match the current node name %s, checked %d Kube API server pods", currentNodeName, len(apiServerPods.Items))
```

Comment:
this sounds like a dangerous situation while it can be completely normal, no?

Reply:
would something less spectacular: `static kube-apiserver pod for node %s not found yet`